image/image.go

```diff
@@ -328,6 +328,20 @@ func makeChannelFromTrack(what, track, defaultChannel string) (string, error) {
 	return mch.Clean().String(), nil
 }
 
+// neededDefaultProviders returns the names of all default-providers for
+// the content plugs that the given snap.Info needs.
+func neededDefaultProviders(info *snap.Info) (cps []string) {
```

Comment:
are we growing many versions of code like this? maybe we can have a shared helper in snap/ ?

Reply:
If this is the case and it's becoming a global helper, then IMHO it makes sense to do what zyga suggested wrt return value (map vs array).

Reply:
I'm looking into a shared version of this code now. It will be a follow-up as it looks like it requires some more work (if sensible at all) because the inputs are slightly different and the usage is also slightly different. We have similar but different code in snapstate (which will also check if a default provider is missing in the snapstate), ifacestate which will get a different input and image.